### PR TITLE
tests: Moved BOARD_INSUFFICIENT_MEMORY to Makefile.ci

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

- Moved `BOARD_INSUFFICIENT_MEMORY` from `Makefile` to `Makefile.ci`
- Formatted the list to have one board per line
- Sorted the lines alphabetically
- Adapted the CI tests to enforce that `BOARD_INSUFFICIENT_MEMORY` does not sneak back into `Makefile`s

### Testing procedure

- Murdock will do the hard ware here, no C code has been changed

### Issues/PRs references

Follow up to https://github.com/RIOT-OS/RIOT/pull/12406

### CODE REVIEW

File: tests/bench_msg_pingpong/Makefile.ci [hunk 1]
```diff
@@ -0,0 +1,2 @@
+BOARD_INSUFFICIENT_MEMORY := \
+    nucleo-f031k6
```
Comment (smlng) [inline tests/bench_msg_pingpong/Makefile.ci]: maybe change this to:

```
    nucleo-f031k6 \
    #
```

to match structure of other files and make additions easy

File: tests/bench_mutex_pingpong/Makefile.ci [hunk 2]
```diff
@@ -0,0 +1,2 @@
+BOARD_INSUFFICIENT_MEMORY := \
+    nucleo-f031k6
```
Comment (smlng) [inline tests/bench_mutex_pingpong/Makefile.ci]: same here, see above

File: tests/bench_thread_flags_pingpong/Makefile.ci [hunk 3]
```diff
@@ -0,0 +1,2 @@
+BOARD_INSUFFICIENT_MEMORY := \
+    nucleo-f031k6
```
Comment (smlng) [inline tests/bench_thread_flags_pingpong/Makefile.ci]: dito

**Comment by maribu** [comment]
@smlng: Thanks for nothing that. I added the missing `\\\n    #` and squashed right away.

**Comment by benpicco** [comment]
Damn, I should have hit the green button on #12411 already 😉  (or just waited a couple more days)

**Comment by smlng** [comment]
CI not happy, some whitespace error, please fix and amend directly.

**Comment by maribu** [comment]
Please wait a moment before merging

**Comment by maribu** [comment]
Using the silver searcher I could use this to find some `Makefile.ci` that were differently typeset:

```
ag -lv -G 'Makefile\.ci' '(#[^\n]*\n)*BOARD_INSUFFICIENT_MEMORY := \\\n(    [a-zA-Z0-9-_]+ \\\n)+    #\n'
```

Maybe some check like this could be added to travis, if someone knows how to do this best?

**Comment by benpicco** [comment]
I think the computer can do a much better job at keeping those lists properly formatted, humans shouldn't be required to meticulously craft those manually.

**Comment by benpicco** [comment]
ACKs don't loose their validity if you push something 😉

**Comment by maribu** [comment]
> ACKs don't loose their validity if you push something :wink:

I think that @MichelRottleuthner told me during FGSN that he wouldn't like it if I would wait for his ACK, than force push completely different stuff and hit the merge button before he can revoke his ACK, and then blame his cause he gave the ACK. (I cannot recall why he told me that so specifically :laughing:)

**Comment by maribu** [comment]
Oha. Merge conflict :cry: I'm on it.

**Comment by benpicco** [comment]
Just squash the last two commits.

**Pull request merged**